app/controllers/reset_passwords_controller.rb

```diff
@@ -0,0 +1,44 @@
+class ResetPasswordsController < ApplicationController
+  skip_before_action :authenticate_user
+  before_action :ensure_no_user
+  before_action :pundit_authorize_user
```

Comment:
As you're checking that there is no user it's probably not needed to also call authorize (authorization = checking if **current user** have access)